momento-sdk/src/intTest/java/momento/sdk/storage/DataTests.java

```diff
@@ -0,0 +1,156 @@
+
+import momento.sdk.BaseTestClass;
+import momento.sdk.PreviewStorageClient;
+import momento.sdk.auth.CredentialProvider;
+import momento.sdk.config.StorageConfigurations;
+import momento.sdk.exceptions.NotFoundException;
+import momento.sdk.responses.storage.data.DeleteResponse;
+import momento.sdk.responses.storage.data.GetResponse;
+import momento.sdk.responses.storage.data.PutResponse;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class DataTests extends BaseTestClass {
+  private static PreviewStorageClient client;
+
+  // TODO can set to the same value as the cache tests
+  // TODO rename env var for clarity to TEST_RESOURCE_NAME or similar
+  private final String storeName = System.getenv("TEST_CACHE_NAME");
+
+  @BeforeAll
+  static void setup() {
+    client =
+        new PreviewStorageClient(
+            CredentialProvider.fromEnvVar("MOMENTO_API_KEY"),
+            StorageConfigurations.Laptop.latest());
+
+    client.createStore(System.getenv("TEST_CACHE_NAME")).join();
+  }
+
+  @AfterAll
+  static void tearDown() {
+    client.close();
+  }
+
+  @Test
+  void getReturnsValueAsStringAfterPut() {
+    final String key = randomString("key");
+    final String value = randomString("value");
+
+    // Successful Set
+    final PutResponse putResponse = client.put(storeName, key, value).join();
+    assertThat(putResponse).isInstanceOf(PutResponse.Success.class);
+
+    // Successful Get
+    final GetResponse getResponse = client.get(storeName, key).join();
+    assertThat(getResponse).isInstanceOf(GetResponse.Success.class);
+    assertThat(getResponse.success().get().value().get().getString().get()).isEqualTo(value);
+  }
+
+  @Test
+  void getReturnsValueAsByteArrayAfterPut() {
+    final String key = randomString("key");
+    final String value = randomString("value");
+
+    // Successful Set
+    final PutResponse putResponse = client.put(storeName, key, value.getBytes()).join();
+    assertThat(putResponse).isInstanceOf(PutResponse.Success.class);
+
+    // Successful Get
+    final GetResponse getResponse = client.get(storeName, key).join();
+    assertThat(getResponse).isInstanceOf(GetResponse.Success.class);
+    assertThat(getResponse.success().get().value().get().getByteArray().get())
+        .isEqualTo(value.getBytes());
+  }
+
+  @Test
+  void getReturnsValueAsLongAfterPut() {
+    final String key = randomString("key");
+    final long value = 42L;
+
+    // Successful Set
+    final PutResponse putResponse = client.put(storeName, key, value).join();
+    assertThat(putResponse).isInstanceOf(PutResponse.Success.class);
+
+    // Successful Get
+    final GetResponse getResponse = client.get(storeName, key).join();
+    assertThat(getResponse).isInstanceOf(GetResponse.Success.class);
+    assertThat(getResponse.success().get().value().get().getLong().get()).isEqualTo(value);
```

Comment:
It's a little long winded but I think each stop in the chain has a purpose. I think we will get asked why we can't remove one link in the chain and either have `success()` directly return `Optional<StorageValue>`, or just expose `.value() -> Optional<StorageValue>` directly on the response parent class. I think in either of those cases we'd be forced to throw an exception if it was not a `Success`. So I think I'm probably leaning towards keeping this as-is for now and getting feedback from others during the bug bash. 

@nand4011 wdyt?